llvm/lib/IR/ProfDataUtils.cpp

```diff
@@ -77,11 +80,25 @@ bool isBranchWeightMD(const MDNode *ProfileData) {
   return isTargetMD(ProfileData, "branch_weights", MinBWOps);
 }
 
+bool isValueProfileMD(const MDNode *ProfileData) {
+  return isTargetMD(ProfileData, "VP", MinVPOps);
+}
+
 bool hasBranchWeightMD(const Instruction &I) {
   auto *ProfileData = I.getMetadata(LLVMContext::MD_prof);
   return isBranchWeightMD(ProfileData);
 }
 
+bool hasCountTypeMD(const Instruction &I) {
+  auto *ProfileData = I.getMetadata(LLVMContext::MD_prof);
+  // Value profiles record count-type information.
+  if (isValueProfileMD(ProfileData))
+    return true;
+  // Conservatively assume non CallBase instruction only get taken/not-taken
+  // branch probability, so not interpret them as count.
+  return isa<CallBase>(I) && !isBranchWeightMD(ProfileData);
```

Comment:
It seems that the check of CallBase is not necessary. Select instruction is another one taking prof MD, and it can only be branch weight type.

If checking callbase is not needed, passing Profile meta data pointer is better.

Reply:
Non-CallBase instructions (e.g., [SwitchInst](https://llvm.org/docs/BranchWeightMetadata.html#switchinst)) could have branch weights with two ops (i.e., one `branch_weight` string and one int), and the helper function `hasCountTypeMD` would be more self-contained to return `false` (tell caller it's not count-type profile) since we know only CallInst or InvokeInst carry count-type values in `branch_weights`.

Reply:
Right, I forgot about SwitchInst with only default case.  (Note that the documentation is out of date. SelectInst also supports branch_weights). LGTM.